### DISCUSSION TITLE
CloudFlare Tunnel + ignoreAgentHashCheck

### DISCUSSION THREAD

**Discussion by AlphaIrri** [Ylianst/MeshCentral#5807] [Issues]
Hello,
I am hoping someone can help me with the following issue.
I have MeshCentral setup on an unRaid docker with CloudFlare Tunnel for external access. This has worked for over six months. This morning all my agents dropped. After troubleshooting, I was able to get agents back online, but I had to add these to my config file:
"trustedProxy": "CloudFlare",
"agentpong": 50,
"browserping": 50,
"ignoreAgentHashCheck": true

Any troubleshooting steps I should try next?
```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "cert": "remote.example.com",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 443,
    "_aliasPort": 443,
    "redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": false,
    "SelfUpdate": false,
    "AllowFraming": "false",
    "WebRTC": "true",
	"trustedProxy": "CloudFlare",
	"agentpong": 50,
	"browserping": 50,
	"ignoreAgentHashCheck": true
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "NewAccounts": "false",
      "_userNameIsEmail": true,
      "certUrl": "https://remote.example.com:443"
      }
  }
}
```

**Comment by openncomp** [comment]
I have the same problem through Cloudflare CDN, out of hundreds of agents, only a few are connected, the rest cannot connect.
I started MeshCentral manually and watched the terminal output, I see that other agents are also connected, but they appear as offline in the interface.
I have a domain connected to Cloudflare, I tried to disconnect it from the CDN, but the problem did not go away.
It seems to me that it is necessary to disconnect the domain from Cloudflare and try a direct connection, perhaps a problem on the CDN side.
